underfs/cosn/pom.xml

```diff
@@ -69,6 +69,37 @@
         <plugin>
           <groupId>org.apache.maven.plugins</groupId>
           <artifactId>maven-shade-plugin</artifactId>
+          <executions>
+            <execution>
+              <id>shade</id>
+              <phase>package</phase>
+              <goals>
+                <goal>shade</goal>
+              </goals>
+              <configuration>
+                <filters>
+                  <filter>
+                    <artifact>*:*</artifact>
+                    <excludes>
+                      <exclude>LICENSE</exclude>
+                      <exclude>META-INF/LICENSE</exclude>
+                      <exclude>META-INF/*.SF</exclude>
+                      <exclude>META-INF/*.DSA</exclude>
+                      <exclude>META-INF/*.RSA</exclude>
+                      <!-- Because this module depends on alluxio-underfs-hdfs, eliminate HDFS factory implementation -->
+                      <exclude>alluxio/underfs/hdfs/HdfsUnderFileSystemFactory.*</exclude>
```

Comment:
If you exclude this, the cosn module would encountered a ClassNotFoundException.

What you want to do is exclude `META-INF/services/alluxio.underfs.UnderFileSystemFactory` only.

Reply:
/ping @elega @jiacheliu3